Doesn't compile with latest git revision of cgmath

I have a program which depends on this crate, as well as the git version of `cgmath` (since it has some functionality not in the stable version).

Recently the `collision` crate stopped compiling with the latest revision of `cgmath`, with errors like this:

```
error[E0277]: the trait bound `<P as cgmath::EuclideanSpace>::Diff: cgmath::UlpsEq<_>` is not satisfied
   --> /home/nate/.cargo/registry/src/github.com-1ecc6299db9ec823/collision-0.20.1/src/algorithm/minkowski/gjk/mod.rs:119:12
    |
119 |         if ulps_eq!(d, P::Diff::zero()) {
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `cgmath::UlpsEq<_>` is not implemented for `<P as cgmath::EuclideanSpace>::Diff`
    |
    = help: the following implementations were found:
              <&'a T as cgmath::UlpsEq>
              <&'a mut T as cgmath::UlpsEq>
    = note: required by `cgmath::Ulps`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Since it only fails to compile with the git version of `cgmath`, this probably isn't a priority, but this will likely need to be fixed once the next version of cgmath is released.

--- a/src/algorithm/minkowski/gjk/mod.rs
+++ b/src/algorithm/minkowski/gjk/mod.rs
@@ -9,6 +9,7 @@ use std::ops::{Neg, Range};
 use cgmath::BaseFloat;
 use cgmath::prelude::*;
 use cgmath::num_traits::NumCast;
+use cgmath::UlpsEq;
 
 use self::simplex::{Simplex, SimplexProcessor2, SimplexProcessor3};
 use crate::{CollisionStrategy, Contact};
@@ -109,7 +110,7 @@ where
         PL: Primitive<Point = P>,
         PR: Primitive<Point = P>,
         SP: SimplexProcessor<Point = P>,
-        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S>,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S> + UlpsEq,
         TL: Transform<P>,
         TR: Transform<P>,
     {
@@ -277,7 +278,7 @@ where
         PL: Primitive<Point = P>,
         PR: Primitive<Point = P>,
         SP: SimplexProcessor<Point = P>,
-        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S>,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S> + UlpsEq,
         TL: Transform<P>,
         TR: Transform<P>,
     {
@@ -365,7 +366,7 @@ where
     ) -> Option<Contact<P>>
     where
         P: EuclideanSpace<Scalar = S>,
-        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S>,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S> + UlpsEq,
         PL: Primitive<Point = P>,
         PR: Primitive<Point = P>,
         TL: Transform<P>,
@@ -415,7 +416,7 @@ where
     ) -> Option<Contact<P>>
     where
         P: EuclideanSpace<Scalar = S>,
-        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S>,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S> + UlpsEq,
         PL: Primitive<Point = P>,
         PR: Primitive<Point = P>,
         TL: Transform<P>,
@@ -479,7 +480,7 @@ where
     ) -> Option<S>
     where
         P: EuclideanSpace<Scalar = S>,
-        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S>,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace + Zero + Array<Element = S> + UlpsEq,
         PL: Primitive<Point = P>,
         PR: Primitive<Point = P>,
         TL: Transform<P>,

